src/app/Table.elm

```diff
@@ -11,93 +11,114 @@ import Html exposing (..)
+type alias State =
     { tablePlayers : TablePlayers
     , myCup : Dice.Cup
+    , currentPlayerId : Int
     }
 
 
-init : Model
-init =
+emptyState : State
+emptyState =
     { tablePlayers = noPlayers
     , myCup = Dice.emptyCup
+    , currentPlayerId = 0
     }
 
 
-fromPlayersAndCup : List ( Int, String, Int ) -> List Int -> Model
-fromPlayersAndCup tuples cup =
-    let
-        tupleToPlayer ( id, name, diceCount ) =
-            { id = id, name = name, diceCount = diceCount }
-    in
-        { tablePlayers = List.map tupleToPlayer tuples
-        , myCup = Dice.cupFromIntList cup
-        }
-
-
 noPlayers : TablePlayers
 noPlayers =
     []
 
 
-testPlayers : TablePlayers
-testPlayers =
-    [ TablePlayer 1 "Joe" 6, TablePlayer 2 "Garrett" 5 ]
-
-
-view : (Msg -> msg) -> Model -> Html msg
-view privateMsg { tablePlayers, myCup } =
+view : TableConfig msg -> State -> Html msg
+view tableConfig { tablePlayers, myCup, currentPlayerId } =
     div []
         [ h3 [] [ text "Players" ]
-        , playersView tablePlayers
-        , liarButton privateMsg
+        , playersView tableConfig tablePlayers currentPlayerId
         , h3 [] [ text "My Cup" ]
         , Dice.viewCup myCup
         ]
 
 
-liarButton privateMsg =
-    button [ onClick <| privateMsg CallLiar ] [ text "Call Liar!" ]
-
-
-playersView : TablePlayers -> Html msg
-playersView tablePlayers =
+playersView : TableConfig msg -> TablePlayers -> Int -> Html msg
+playersView tableConfig tablePlayers currentPlayerId =
     div [] <|
         List.map
-            playerView
+            (\player -> playerView tableConfig player (currentPlayerId == player.id))
             tablePlayers
 
 
-playerView : TablePlayer -> Html msg
-playerView { id, name, diceCount } =
-    div
-        [ style
-            [ ( "border", "1px solid green" )
-            , ( "width", "200px" )
-            , ( "padding", "10px" )
-            , ( "margin", "8px" )
-            , ( "display", "inline-block" )
+playerView : TableConfig msg -> TablePlayer -> Bool -> Html msg
+playerView { callLiar } { id, name, diceCount } isCurrentPlayer =
+    let
+        backgroundColourOfCurrentPlayer =
+            if isCurrentPlayer then
+                "#0f0"
+            else
+                "#cfc"
+
+        optionalLiarButtonDiv =
+            if isCurrentPlayer then
+                [ div [ onClick <| callLiar id ] [ liarButton name ] ]
+            else
+                []
+    in
+        div
+            [ style
+                [ ( "border", "5px solid #262" )
+                , ( "width", "200px" )
+                , ( "background-color", backgroundColourOfCurrentPlayer )
+                , ( "padding", "10px" )
+                , ( "margin", "8px" )
+                , ( "display", "inline-block" )
+                ]
             ]
-        ]
-        [ div [] [ text name ]
-        , div [] [ text <| Dice.describeCount diceCount ]
+            ([ div [ iconStyle ] [ identicon "50px" name ]
```

Comment:
cool, it uses the name which is a string as a hashing function? to create a unique icon avatar :)